### PR TITLE
[APIView] Misc Java-specific improvements

### PR DESCRIPTION
**Pull request by JonathanGiles**
1. Add new diagnostic rule that checks if URLs have locale language tags in them (e.g. 'en-us').

### CODE REVIEW

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/NoLocalesInJavadocUrlDiagnosticRule.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.getPublicOrProtectedMethods;
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.makeId;
+import static com.azure.tools.apiview.processor.model.DiagnosticKind.WARNING;
+
+public class NoLocalesInJavadocUrlDiagnosticRule implements DiagnosticRule {
+
+    private final String[] languageTags;
+
+    public NoLocalesInJavadocUrlDiagnosticRule() {
+        languageTags = Arrays.stream(Locale.getAvailableLocales())
```
Comment (alzimmermsft) [inline src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/NoLocalesInJavadocUrlDiagnosticRule.java:24]: Would it be better to make this into a regex?

```java
StringBuilder patternBuilder = new StringBuilder(".*?\\/(");

Locale[] availableLocales = Locale.getAvailableLocales();
for (int i = 0; i < availableLocales.length; i++) {
    if (i > 0) {
        patternBuilder.append('|');
    }
    
    patternBuilder.append(availableLocale[i].toLanguageTag());
}

patternBuilder.append(")\\/.*");

localePattern = Pattern.compile(patternBuilder.toString(), Pattern.CASE_INSENSITIVE);

Comment (JonathanGiles) [reply src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/NoLocalesInJavadocUrlDiagnosticRule.java:24]: regex is always better - I just pity the next person who has to maintain the code! :)

Comment (JonathanGiles) [reply src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/NoLocalesInJavadocUrlDiagnosticRule.java:24]: In this case it is nice to iterate over each, as then we know which language tag is the issue and we can report that to the user to make search / replace easier.

**Review by mssfang — APPROVED** [review]

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/NoLocalesInJavadocUrlDiagnosticRule.java (hunk 1 above)
Comment (alzimmermsft) [reply src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/NoLocalesInJavadocUrlDiagnosticRule.java:24]: Agreed, realized after I wrote this that the current pattern is better as it allows multiple instances to be marked.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**